src/chunk.c

```diff
@@ -2859,11 +2845,11 @@ ts_chunk_create_fks(Chunk *const chunk)
 	ASSERT_IS_VALID_CHUNK(chunk);
 
 	rel = table_open(chunk->hypertable_relid, AccessShareLock);
-	fks = copy_fk_list_from_cache(RelationGetFKeyListCompat(rel));
+	fks = RelationGetFKeyList(rel);
```

Comment:
The result needs be copied. This is why the copy function was present.

Reply:
good catch, expanded the wrong macro there

Reply:
It happened that I remember this code :)
I haven't reviewed much of other code as it is still in draft, but this one unlikely to be caught by our tests.